src/util/expr.cpp

```diff
@@ -279,14 +279,80 @@ void exprt::visit(std::function<void(const exprt &)> visitor) const
   }
 }
 
+void exprt::visit_post(std::function<void(exprt &)> visitor)
+{
+  struct stack_entryt
+  {
+    exprt *e;
+    bool operands_pushed;
+    explicit stack_entryt(exprt *_e) : e(_e), operands_pushed(false)
+    {
+    }
+  };
+
+  std::stack<stack_entryt> stack;
+
+  stack.emplace(this);
+
+  while(!stack.empty())
+  {
+    auto &top = stack.top();
+    if(top.operands_pushed)
+    {
+      visitor(*top.e);
+      stack.pop();
+    }
+    else
+    {
+      // do modification before pushing in case 'top' isn't stable
+      top.operands_pushed = true;
+      for(auto &op : top.e->operands())
+        stack.emplace(&op);
+    }
+  }
+}
+
+void exprt::visit_post(std::function<void(const exprt &)> visitor) const
```

Comment:
There is a lot of duplication with the non-const version, which is not ideal. Maybe this could be avoided using templates?

Reply:
done